### PR TITLE
Add apply lambda 

### PR DESCRIPTION
**Pull request by Lyxal**
Provides somewhat of a fix to #336. I've had this in mind for 2 months now.

### CODE REVIEW

File: documents/knowledge/elements.yaml [hunk 1]
```diff
@@ -39,17 +39,9 @@
...
-  description: Returns the same as logical and but with reversed arguments.
-  overloads:
-    any-any: b and a
-  vectorise: false
-  tests:
-    - "[0, 0] : 0"
-    - '["", 1] : ""'
-    - "[[1,2,3], 0] : 0"
-    - "[1, 2] : 1"
+  name: Apply Lambda
```
Comment (cgccuser) [inline documents/knowledge/elements.yaml:42]: Poggers, might want to also call it foreach

File: documents/knowledge/elements.yaml [hunk 2]
```diff
@@ -39,17 +39,9 @@
...
+  arity: 1
+  description: Like a mapping lambda, but the results are evaluated immediantly, instead of being lazily evaluated
```
Comment (cgccuser) [inline documents/knowledge/elements.yaml:44]: Immediately

**Review by cgccuser — CHANGES_REQUESTED** [review]
Minor changes but this is as good as approved

**Review by cgccuser — APPROVED** [review]

**Review by PyGamer0 — APPROVED** [review]
ok

**Pull request merged**